codecov/__init__.py

```diff
@@ -719,8 +719,16 @@ def main(*argv, **kwargs):
                         res = res.text.strip().split()
                         result, upload_url = res[0], res[1]
 
+                        # Handle reports encoding for Python 2 and 3
+                        try:
+                            basestring
+                        except NameError:
+                            basestring = str
+                        if isinstance(reports, basestring):
```

Comment:
Almost good.

For Python 2 you want `if isinstance(reports, unicode)` (not `(str, unicode)`), 
for python 3 you want `if isinstance(reports, str)`.

So the correct fix would be:

```python
try:
    unicode
except NameError:
    # Must be Python 3, make an alias for str.
    unicode = str
if isinstance(reports, unicode):
    reports = reports.encode('utf-8')
```

However, Python 2 has `bytes` being an alias for `str`. So you can do:

```python
if not isinstance(reports, bytes):
    reports = reports.encode('utf-8')
```